### ISSUE TITLE
Maven Compiler/Lombok Error When Building App Container Images

### ISSUE THREAD

**Issue by michael-intindola** [GoogleCloudPlatform/fda-mystudies#3409]
**Describe the bug**
I attempted to upgrade a few of my instances of "FDA MyStudies on GCP" to v2.0.3, and the Cloud Build pipelines in the "`$PREFIX-$ENV-apps`" project are all failing with the following error:

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project common-service: Fatal error compiling: java.lang.IllegalAccessError: class lombok.javac.apt.LombokProcessor (in unnamed module @0x5a720cdf) cannot access class com.sun.tools.javac.processing.JavacProcessingEnvironment (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.processing to unnamed module @0x5a720cdf -> [Help 1]`

I updated one of my instances 8 days ago and it worked fine. I realized the error yesterday when I attempted to upgrade another one of my instances. Out of curiosity, I attempted to re-build the instance that worked last week, and that one is failing now also. This makes me think it is an issue with a dependency.

This [issue](https://github.com/rzwitserloot/lombok/issues/2681) in the Lombok repository makes me believe it is related to a Lombok fix for JDK 16 approximately 5 days ago, but I am not completely certain.

**To Reproduce**
1. Pull the latest code (v2.0.3) to my local repository and push the updated code to my remote repository
2. The checks start running for `auth-server`, `study-builder`, `participant-manager-datastore`, `participant-datastore-enroll-mgmt-module`, `participant-datastore-user-mgmt-module`, `participant-datastore-consent-mgmt-module`, `response-datastore`, and `study-datastore` and they all fail with the above mentioned error

**Screenshots**
<img width="1789" alt="Screen Shot 2021-03-24 at 9 57 06 AM" src="https://user-images.githubusercontent.com/46120033/112322526-5cc7be80-8c87-11eb-9e62-68f7eb676af1.png">

**Comment by moschetti** [comment]
@dhanyak-btc could your team look at this?

**Comment by dhanyak-btc** [comment]
@moschetti We've tested Study Builder and Study Metadata on Java 7 and Spring Boot based components on Java 8. Upgrading to higher version of Java may cause build failures due to incompatible dependencies. Please let me know which version of Java we need to upgrade, we'll implement in next milestone.

**Comment by dhanyak-btc** [comment]
@moschetti we are looking into this issue.

**Comment by michael-intindola** [comment]
@dhanyak-btc Have you confirmed that this is also an issue for you? I was unsure if this was an issue with something on my end or with a dependency.

**Comment by dhanyak-btc** [comment]
@michael-intindola We didn't had this issue in past deployments, we noticed this error today only. We'll add a comment here once the issue is fixed.

**Comment by smunini** [comment]
I have encountered this issue beginning today as well:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project common-service: Fatal error compiling: java.lang.IllegalAccessError: class lombok.javac.apt.LombokProcessor (in unnamed module @0x377bc8da) cannot access class com.sun.tools.javac.processing.JavacProcessingEnvironment (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.processing to unnamed module @0x377bc8da -> [Help 1]

**Comment by smunini** [comment]
Is there a way to downgrade to an earlier JDK 16 version that was used a few days ago?  Where would one find that configuration in this project?

**Comment by dhanyak-btc** [comment]
@smunini @michael-intindola Looks like `--illegal-access` default value changed from `permit `to `deny`, please refer to the description section in [JEP 396: Strongly Encapsulate JDK Internals by Default](https://openjdk.java.net/jeps/396) for more details. 

I've added below code in pom.xml of `auth-server`,` participant-manager-datastore`,` participant-datastore-enroll-mgmt-module`, `participant-datastore-user-mgmt-module`, `participant-datastore-consent-mgmt-module`, `response-datastore` it worked fine. We'll add a comment here once the `study builder` and `study datastore `issues are fixed.

```
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <configuration>
          <source>11</source>
          <target>11</target>
          <fork>true</fork>
          <compilerArgs>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.jvm=ALL-UNNAMED</arg>
          </compilerArgs>
          <annotationProcessorPaths>
            <path>
              <groupId>org.projectlombok</groupId>
              <artifactId>lombok</artifactId>
              <version>1.18.12</version>
            </path>
          </annotationProcessorPaths>
        </configuration>
      </plugin>
```

**Comment by smunini** [comment]
@dhanyak-btc , thank you for your response.  I found I needed to add the above plugin to all of the poms in those projects and not just the root poms.  Here is a list of the ones I touched:

auth-server/oauth-scim-service/pom.xml
auth-server/pom.xml
common-modules/common-tests/pom.xml
common-modules/common-service/pom.xml
common-modules/pom.xml
participant-datastore/consent-mgmt-module/consent-mgmt/pom.xml
participant-datastore/consent-mgmt-module/pom.xml
participant-datastore/enroll-mgmt-module/enroll-mgmt/pom.xml
participant-datastore/enroll-mgmt-module/pom.xml
participant-datastore/user-mgmt-module/user-mgmt/pom.xml
participant-manager-datastore/participant-manager-service/pom.xml
participant-manager-datastore/pom.xml
response-datastore/pom.xml
response-datastore/response-server-service/pom.xml

I'll stay tuned about `study-builder` and `study-datasource`.  Thanks!

**Comment by dhanyak-btc** [comment]
@smunini For `study-datastore`, change the `maven-war-plugin` and `maven-compiler-plugin` version to `3.8.1`. Also, add the `<configuration>` as given below for `maven-compiler-plugin`.

```
<plugin>
        <artifactId>maven-war-plugin</artifactId>
        <version>3.3.1</version>
        <configuration>
          <packagingExcludes>WEB-INF/web.xml</packagingExcludes>
          <packagingExcludes>META-INF/context.xml</packagingExcludes>
          <failOnMissingWebXml>false</failOnMissingWebXml>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>3.8.1</version>
        <configuration>
          <webXml>WEB-INF/web.xml</webXml>
          <source>11</source>
          <target>11</target>
          <fork>true</fork>
          <compilerArgs>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.jvm=ALL-UNNAMED</arg>
          </compilerArgs>
          <annotationProcessorPaths>
            <path>
              <groupId>org.projectlombok</groupId>
              <artifactId>lombok</artifactId>
              <version>1.18.12</version>
            </path>
          </annotationProcessorPaths>
        </configuration>
      </plugin>
```

**Comment by smunini** [comment]
That worked thank you @dhanyak-btc !

**Comment by smunini** [comment]
@dhanyak-btc , did you happen to also have a fix for `study-builder`?

**Comment by smunini** [comment]
Here is a potential temporary solution, however I'm running into a new issue with the tests below:

https://github.com/PatrykLeczycki/tgtg-spring-boot/commit/eed5763906718987cec41642a1e5c21d58f7049e

Running com.fdahpstudydesigner.controller.StudyActiveTasksControllerTest
2021-03-28 00:38:02,442 ERROR [main] (TestContextManager.java:309) - Caught exception while allowing TestExecutionListener [org.springframework.test.context.support.DependencyInjectionTestExecutionListener@34a91445] to prepare test instance [com.fdahpstudydesigner.controller.StudyActiveTasksControllerTest@4a09f5b]
java.lang.IllegalStateException: Failed to load ApplicationContext
	at org.springframework.test.context.CacheAwareContextLoaderDelegate.loadContext(CacheAwareContextLoaderDelegate.java:99)
	at org.springframework.test.context.TestContext.getApplicationContext(TestContext.java:122)
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.injectDependencies(DependencyInjectionTestExecutionListener.java:109)
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.prepareTestInstance(DependencyInjectionTestExecutionListener.java:75)
	at org.springframework.test.context.TestContextManager.prepareTestInstance(TestContextManager.java:307)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.createTest(SpringJUnit4ClassRunner.java:211)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner$1.runReflectiveCall(SpringJUnit4ClassRunner.java:288)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.methodBlock(SpringJUnit4ClassRunner.java:284)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:231)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:88)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:71)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:236)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:174)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.IllegalStateException: Cannot load configuration class: com.fdahpstudydesigner.config.WebAppTestConfig
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.enhanceConfigurationClasses(ConfigurationClassPostProcessor.java:368)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanFactory(ConfigurationClassPostProcessor.java:244)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:694)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:633)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:461)
	at org.springframework.test.context.web.AbstractGenericWebContextLoader.loadContext(AbstractGenericWebContextLoader.java:128)
	at org.springframework.test.context.web.AbstractGenericWebContextLoader.loadContext(AbstractGenericWebContextLoader.java:60)
	at org.springframework.test.context.support.AbstractDelegatingSmartContextLoader.delegateLoading(AbstractDelegatingSmartContextLoader.java:100)
	at org.springframework.test.context.support.AbstractDelegatingSmartContextLoader.loadContext(AbstractDelegatingSmartContextLoader.java:248)
	at org.springframework.test.context.CacheAwareContextLoaderDelegate.loadContextInternal(CacheAwareContextLoaderDelegate.java:64)
	at org.springframework.test.context.CacheAwareContextLoaderDelegate.loadContext(CacheAwareContextLoaderDelegate.java:91)
	... 31 more
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.springframework.cglib.proxy.Enhancer
	at org.springframework.context.annotation.ConfigurationClassEnhancer.newEnhancer(ConfigurationClassEnhancer.java:110)
	at org.springframework.context.annotation.ConfigurationClassEnhancer.enhance(ConfigurationClassEnhancer.java:98)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.enhanceConfigurationClasses(ConfigurationClassPostProcessor.java:358)
	... 41 more

**Comment by smunini** [comment]
@dhanyak-btc, Since that change, `study-datastore` is now in an unhealthy state with lots of these in the log:

2021-03-28 19:54:25.114 EDT28-Mar-2021 23:54:25.114 INFO [http-nio-8080-exec-5] com.sun.jersey.api.core.PackagesResourceConfig.init Scanning for root resource and provider classes in the packages:
Error
2021-03-28 19:54:25.114 EDT com.hphc.mystudies.service
Error
2021-03-28 19:54:25.114 EDT com.hphc.mystudies.dto
Error
2021-03-28 19:54:25.114 EDT com.hphc.mystudies.corn
Error
2021-03-28 19:54:25.114 EDT org.codehaus.jackson.jaxrs

Is it possible to have two versions of the Jersey jars as described in StackOverflow post? https://stackoverflow.com/questions/48441701/tomcat-log-not-showing-com-sun-jersey-api-core-scanningresourceconfig-logclasses

**Comment by smunini** [comment]
@dhanyak-btc , for `study-datastore`, keeping your maven-compiler-plugin config but changing to Java 1.7 cleared up the Jersey issue:

<source>1.7</source>
<target>1.7</target>

**Comment by michael-intindola** [comment]
@smunini can you please show the configuration changes you made in `study-builder` and `study-datastore` pom files to get Study Builder building without error?

**Comment by smunini** [comment]
@michael-intindola , for `study-datastore`, I made this change to the pom as suggested by @dhanyak-btc but using Java 7 and not 11.

	     <plugin>
        <artifactId>maven-war-plugin</artifactId>
        <version>3.3.1</version>
        <configuration>
          <packagingExcludes>WEB-INF/web.xml</packagingExcludes>
          <packagingExcludes>META-INF/context.xml</packagingExcludes>
          <failOnMissingWebXml>false</failOnMissingWebXml>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>3.8.1</version>
        <configuration>
          <webXml>WEB-INF/web.xml</webXml>
          <source>1.7</source>
          <target>1.7</target>
          <fork>true</fork>
          <compilerArgs>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
            <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.jvm=ALL-UNNAMED</arg>
          </compilerArgs>
          <annotationProcessorPaths>
            <path>
              <groupId>org.projectlombok</groupId>
              <artifactId>lombok</artifactId>
              <version>1.18.12</version>
            </path>
          </annotationProcessorPaths>
        </configuration>
      </plugin>


For `study-builder`, it is still not functional.  I tried making a similar change, but that did not work.  I tried this change (https://github.com/GoogleCloudPlatform/fda-mystudies/issues/3409#issuecomment-808824476) but couldn't work around the test failures.  I also tried removing lombok completely, and manually added the getters, setters and constructors, but that didn't work either.  I'm still searching for a solution to get `study-builder` to work.

**Comment by michael-intindola** [comment]
Thank you @smunini . We are on the same page then. Please let me know if you figure anything out with `study-builder` and I will do the same.

**Comment by smunini** [comment]
Will do @michael-intindola

**Comment by dhanyak-btc** [comment]
@smunini @michael-intindola Sorry, I was on sick leave, so didn't respond to your comments. Above fix didn't work for `study-builder` and `study-datastore` so we replaced` jib-maven-plugin` with `Dockerfile`, please refer to PR #3427 for the fix.

**Comment by dhanyak-btc** [comment]
Fixed in PR #3427
